### DISCUSSION TITLE
"Re-sourcing config is not supported", trying to using everforest theme with a custom config.

### DISCUSSION THREAD

**Discussion by Virayen** [LazyVim/LazyVim#4213] [Q&A]
Hey guys, Im trying to use the  everforest theme (https://github.com/neanias/everforest-nvim?tab=readme-ov-file#usage). 
And have edited the everforest.lua file according to what they describe, but I get an error like this, 
![image](https://github.com/user-attachments/assets/be8d367f-1ffe-40ef-824c-aa22e534afbc)
 My file, 
 
![image](https://github.com/user-attachments/assets/6d6ca558-8b21-4c24-941f-eeec2afc9e49)

The reason Im doing this is so I can use the 'hard' version of everforest, shown in the github, and not just 
![image](https://github.com/user-attachments/assets/04b113fc-a2dc-42a7-84a0-58d4d01c5c3a)
which loads the default config with the medium version of everforest.

**Comment by dpetka2001** [comment]
In `everforest.lua` you are doing `require("lazy").setup` and that's why you're seeing the error. `lazy` has already been required in `config.lazy` in LazyVim. Just put the options you want in the `everforest` spec itself.

**Reply by Virayen** [reply]
Thank you! I should read a little it seems

**Comment by LorenzoBettini** [comment]
Maybe I'm missing something.

I return the spec as follows:

```lua
  {
    "neanias/everforest-nvim",
    opts = {
      background = "hard", -- "soft", "medium", "hard"
    },
  },
```

But when restarting Neovim I get:

```
lazy.nvim Lua module not found for config of everforest-nvim. Please use a `config()` function instead
```

So what is the correct way to do that?

**Reply by dpetka2001** [reply]
This happens because of how `lazy.nvim` (the package manager) tries to determine the main Lua module of a plugin. Usually it takes whatever is in the plugin name and splits it across `.` (dot punctuation). 

If the plugin was named `everforest.nvim`, then it would correctly determine the main Lua module of the plugin to be `everforest`. But because the plugin is named `everforest-nvim` (dash instead of dot) it doesn't do any splitting and expects that the main Lua module will also be called `everforest-nvim`, but the plugin's main Lua module is called `everforest` instead.

To work around this, you can do something like
```lua
{
    "neanias/everforest-nvim",
    main = "everforest",
    lazy = false,
    opts = {
      background = "hard", -- "soft", "medium", "hard"
    },
  },
```
This will explicitly tell `lazy.nvim` that the main Lua module of the plugin is to be called `everforest` and so `lazy.nvim` will not go on with automatically trying to determine the main Lua module (which expects it to be `everforest-nvim` because of the plugin's naming convention).

**Reply by LorenzoBettini** [reply]
Thank you @dpetka2001 !
That was it! :)
